### PR TITLE
Fixed panic when no server_url is provided

### PR DESCRIPTION
**Pull request by notanthony**
added option to just provide tenant domain, added a validator that makes sure exactly one of [server_url, tenanat_domain] values are set

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -69,9 +87,10 @@ func (p *ConductoroneProvider) Configure(ctx context.Context, req provider.Confi
 	ServerURL := data.ServerURL.ValueString()
 	ClientID := data.ClientID.ValueString()
 	ClientSecret := data.ClientSecret.ValueString()
+	TenantDomain := data.TenantDomain.ValueString()
 
 	if ServerURL == "" {
-		ServerURL = "https://{tenantDomain}.conductor.one"
+		ServerURL = fmt.Sprintf("https://%s.conductor.one", TenantDomain)
```
Comment (pquerna) [inline internal/provider/provider.go:93]: please use a URL builder

File: internal/provider/provider.go [hunk 2]
```diff
@@ -69,9 +87,10 @@ func (p *ConductoroneProvider) Configure(ctx context.Context, req provider.Confi
 	ServerURL := data.ServerURL.ValueString()
 	ClientID := data.ClientID.ValueString()
 	ClientSecret := data.ClientSecret.ValueString()
+	TenantDomain := data.TenantDomain.ValueString()
```
Comment (pquerna) [inline internal/provider/provider.go:90]: how does this work if I only set ClientID and ClientSecret?

We should be infering the serverurl/tenant domain just based on the client id in that case

File: internal/provider/provider.go (hunk 1 above)
Comment (notanthony) [reply internal/provider/provider.go]: https://github.com/ConductorOne/terraform-provider-conductorone/blob/4c3c783bfa75bc61bc8a13eea3bc3803e676eab0/internal/sdk/extra_sdk_options.go#L24C14-L41

Below we call sdk.WithTenantCustom(serverURL) which called normalizeTenant(input) which uses a URL builder

File: internal/provider/provider.go [hunk 3]
```diff
@@ -69,9 +77,14 @@ func (p *ConductoroneProvider) Configure(ctx context.Context, req provider.Confi
...
 
 	if ServerURL == "" {
-		ServerURL = "https://{tenantDomain}.conductor.one"
+		if TenantDomain == "" {
+			url := strings.Split(ClientID, "@")[1]
+			TenantDomain = strings.Split(url, ".")[0]
```
Comment (pquerna) [inline internal/provider/provider.go:85]: we should use the SDK function for this logic?  doesn't it fully handle splitting the Client ID?

File: internal/provider/provider.go [hunk 4]
```diff
@@ -69,9 +77,15 @@ func (p *ConductoroneProvider) Configure(ctx context.Context, req provider.Confi
...
-		ServerURL = "https://{tenantDomain}.conductor.one"
+		if TenantDomain == "" {
+			url := strings.Split(ClientID, "@")[1]
+			ServerURL = fmt.Sprintf("https://%s", url)
+		} else {
+			ServerURL = fmt.Sprintf("https://%s.conductor.one", TenantDomain)
+		}
 	}
 
 	opt, err := sdk.WithTenantCustom(ServerURL)
```
Comment (pquerna) [inline internal/provider/provider.go:91]: like this logic is backwards, isn't it?

Shouldn't we only set this option, IF TenantDomain or ServerURL is set.  And then let the SDK fallback to parsing the client ID which it already doesn't internally?

**Pull request closed without merging**